test/acceptance/features/bindAppToService.feature

```diff
@@ -305,3 +304,42 @@ Feature: Bind an application to a service
         Then jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding Request "binding-request-a-d-c" should be changed to "True"
         And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding Request "binding-request-a-d-c" should be changed to "True"
         And Secret "binding-request-a-d-c" contains "SOME_KEY" key with value "SOME_VALUE:5432:db-demo-a-d-c"
+
+    Scenario: Service Binding Request to fetch data from scalar, sequence, map data structures from Service CR
```

Comment:
As mentioned in [the previous review round](https://github.com/redhat-developer/service-binding-operator/pull/655#discussion_r485029702), scenario description should reflect impact of a given user input (context) to the system. The particular scenario is about injecting scalar, sequence and map values into the application. In order to achieve that we use custom env var functionality. What is unclear to me is if this can be done using appropriate annotations as well, but it is ok to explore it in another PR, since the focus here is on custom env variables.

Anyhow, the current description does not mention the impact on the system. How about something like this:

```
Scalar, sequence and map values from service resource
are injected into application using custom env variables
```

In order to keep focus better, it is strongly recommended to split this scenario into multiple ones, for each source value type.

My understand is that this PR is follow-up of https://github.com/redhat-developer/service-binding-operator/pull/602 where we [documented](https://github.com/redhat-developer/service-binding-operator/tree/master/examples/java_postgresql_customvar) the example that will be translated into one or more acceptance tests that perform the same things suggested to users, but in an automated way.

The current scenario does not perform the checks stated in [the example](https://github.com/redhat-developer/service-binding-operator/blob/master/examples/java_postgresql_customvar/README.md#check-the-environment-variable-creation-in-the-application-pod), but it should. The existence of each expected variable and its value should be checked. Without that, we cannot be sure if the code does what is expected.

Acceptance tests treat system as black box and should perform checks that are important from user perspective. In this case, user want to know how the injected env vars look like, in particular how to specify them and what the naming convention.